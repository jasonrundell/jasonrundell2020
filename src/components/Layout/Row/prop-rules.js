const justifyRules = [
  'start',
  'end',
  'center',
  'spaceAround',
  'spaceBetween',
  'spaceEvenly',
]

export default { justifyRules }
