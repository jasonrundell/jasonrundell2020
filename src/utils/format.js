/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
