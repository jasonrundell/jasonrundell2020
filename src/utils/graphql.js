export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
