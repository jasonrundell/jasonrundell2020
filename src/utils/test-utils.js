export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
