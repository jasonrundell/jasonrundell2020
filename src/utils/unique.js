const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
