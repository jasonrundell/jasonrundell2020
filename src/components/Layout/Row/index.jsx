import React from 'react'
import PropTypes from 'prop-types'

import Rules from './prop-rules'

import { classNames } from '../../../utils/css'

import { root } from './Row.module.scss'

const { justifyRules } = Rules

const Row = ({ justify, children }) => {
  const classes = classNames(styles.root, styles[justify])

  return <div className={classes}>{children}</div>
}

Row.defaultProps = {
  justify: justifyRules[0],
}

Row.propTypes = {
  justify: PropTypes.oneOf(justifyRules),
  children: PropTypes.any.isRequired,
}

export default Row
