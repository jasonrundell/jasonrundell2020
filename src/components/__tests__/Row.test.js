import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Row from '../Layout/Row'
import Rules from '../Layout/Row/prop-rules'

const { justifyRules } = Rules

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Row component', () => {
  it('Row renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <Row>Row component</Row>
          <Row justify={justifyRules[0]}>
            Row component justify={justifyRules[0]}
          </Row>
          <Row justify={justifyRules[justifyRules.length - 1]}>
            Row component justify={justifyRules[justifyRules.length - 1]}
          </Row>
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Row has required children prop', () => {
    act(() => {
      render(<Row>Row component</Row>, container)
    })
    expect(container.textContent).toBe('Row component')
  })
})
