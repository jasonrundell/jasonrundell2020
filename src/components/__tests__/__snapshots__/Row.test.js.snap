// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Row renders correctly 1`] = `
<div>
  <div
    className="root start"
  >
    Row component
  </div>
  <div
    className="root start"
  >
    Row component justify=
    start
  </div>
  <div
    className="root spaceEvenly"
  >
    Row component justify=
    spaceEvenly
  </div>
</div>
`;
